### PR TITLE
feat(types): add local structs for assets

### PR DESCRIPTION
**Pull request by GoodluckH**

### CODE REVIEW

File: server/src/structs.rs [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+// =====================================================================
+
+/// A struct representing an aircraft.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct Aircraft {
+    pub id: Uuid,
+    pub name: Option<String>,
+    pub group_id: Option<Uuid>,
+    pub owner: Uuid,
+    pub delegatee: Option<Uuid>,
```
Comment (amsmith-pro) [inline server/src/structs.rs]: I think maybe groups can be delegated but not individual assets

A group of asset size one can be delegated

File: server/src/structs.rs [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    pub created_at: SystemTime,
+    pub updated_at: Option<SystemTime>,
+    /// If the vector is empty, the aircraft is available to everyone.
+    /// Otherwise, the aircraft is only available to the clients in the
+    /// vector.
+    pub restricted_to: Vec<Uuid>,
```
Comment (amsmith-pro) [inline server/src/structs.rs]: is `whitelist` fine?

noted these are temporary

File: server/src/structs.rs [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+///
+/// A vertiport is an airport that is used for vertical takeoff and
+/// landing (VTOL) aircraft. A vertiport may have one or more vertipads.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct Vertiport {
+    pub id: Uuid,
+    pub name: String,
+    pub group_id: Option<Uuid>,
+    pub owner: Uuid,
+    pub delegatee: Option<Uuid>,
```
Comment (amsmith-pro) [inline server/src/structs.rs]: Same comment about groups being delegated, not individual assets

Comment (GoodluckH) [reply server/src/structs.rs]: [21cddfd](https://github.com/Arrow-air/svc-assets/pull/2/commits/21cddfde26697af3ef9860ef25210cb3acff23e2)

File: server/src/structs.rs (hunk 2 above)
Comment (GoodluckH) [reply server/src/structs.rs]: [21cddfd](https://github.com/Arrow-air/svc-assets/pull/2/commits/21cddfde26697af3ef9860ef25210cb3acff23e2)

File: server/src/structs.rs (hunk 1 above)
Comment (GoodluckH) [reply server/src/structs.rs]: [21cddfd](https://github.com/Arrow-air/svc-assets/pull/2/commits/21cddfde26697af3ef9860ef25210cb3acff23e2)

**Comment by GoodluckH** [comment]
Squashed those commits into one to fix some failed github actions.

Refactored the code to have a Basics struct that's shared by all asset types -- reduces repetitive codes.

Also wrote unit tests.

**Review by amsmith-pro — APPROVED** [review]

**Pull request merged**